ibft/ibft_running_instance.go

```diff
@@ -0,0 +1,92 @@
+package ibft
+
+import (
+	"github.com/bloxapp/ssv/ibft/proto"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// startInstanceWithOptions will start an iBFT instance with the provided options.
+// Does not pre-check instance validity and start validity!
+func (i *ibftImpl) startInstanceWithOptions(instanceOpts InstanceOptions, value []byte) (chan *InstanceResult, error) {
+	i.currentInstance = NewInstance(instanceOpts)
+	i.currentInstance.Init()
+	stageChan := i.currentInstance.GetStageChan()
+
+	// reset leader seed for sequence
+	if err := i.currentInstance.Start(value); err != nil {
+		return nil, errors.WithMessage(err, "could not start iBFT instance")
+	}
+
+	i.CurrentInstanceResultChan = make(chan *InstanceResult)
+
+	// main instance callback loop
+	go func() {
+		for {
+			switch stage := <-stageChan; stage {
+			case proto.RoundState_Prepare:
+				if err := i.ibftStorage.SaveCurrentInstance(i.GetIdentifier(), i.currentInstance.State); err != nil {
+					i.logger.Error("could not save prepare msg to storage", zap.Error(err))
+					return
+				}
+			case proto.RoundState_Decided:
+				agg, err := i.currentInstance.CommittedAggregatedMsg()
+				if err != nil {
+					i.pushAndCloseInstanceResultChan(&InstanceResult{
+						Decided: true,
```

Comment:
looks like we passing decided:true in case of error, should be false

Reply:
it is decided.. just has an error